source/compiler-core/slang-json-parser.cpp

```diff
@@ -0,0 +1,384 @@
+// slang-json-parser.cpp
+#include "slang-json-parser.h"
+
+#include "slang-json-diagnostics.h"
+
+/*
+https://www.json.org/json-en.html
+*/
+
+namespace Slang {
+
+SlangResult JSONParser::_parseObject()
+{
+    SLANG_RETURN_ON_FAIL(m_lexer->expect(JSONTokenType::LBrace));
+
+    m_listener->startObject();
+
+    if (m_lexer->advanceIf(JSONTokenType::RBrace))
+    {
+        m_listener->endObject();
+        return SLANG_OK;
+    }
+
+    while (true)
+    {
+        JSONToken keyToken;
+        SLANG_RETURN_ON_FAIL(m_lexer->expect(JSONTokenType::StringLiteral, keyToken));
+        m_listener->addLexemeKey(m_lexer->getLexeme(keyToken));
+
+        SLANG_RETURN_ON_FAIL(m_lexer->expect(JSONTokenType::Colon));
+
+        SLANG_RETURN_ON_FAIL(_parseValue());
+        if (m_lexer->advanceIf(JSONTokenType::Comma))
+        {
+            continue;
+        }
```

Comment:
This gets into the same slippery slope as adding comment support: I'd personally like it if our JSON parser could work robustly when objects or arrays include a trailing comma. We can make the JSON lexer/parser have some flag(s) to mark extensions as disabled and go into a pedantic mode where it diagnoses them as errors. For ordinary development, I imagine that trailing commas will be the most valuable feature after comments.